File: examples/python/load_extension.py
import sqlite3
import os

# Path to your compiled extension (.dylib for macOS/iOS, .so for linux/Android, .dll for Windows)
# Using `https://github.com/sqliteai/sqlite-js` extension as an example
EXTENSION_PATH = os.path.abspath("js")

# Connect to SQLite and enable extension loading
conn = sqlite3.connect(":memory:")
conn.enable_load_extension(True)

# Load the extension
try:
    conn.load_extension(EXTENSION_PATH)
    print("Extension loaded successfully.")
except sqlite3.OperationalError as e:
    print(f"Failed to load extension: {e}")

conn.enable_load_extension(False)

# Optionally test it (e.g., call a custom SQL function)
cursor = conn.execute("SELECT js_version();")
print(cursor.fetchone())

File: examples/php/load_extension.php
<?php
ini_set("sqlite3.extension_dir", "/usr/lib/php/20240924");

/**
 * Install the official SQLite package for PHP in your system.
 * Then, in the PHP ini, uncomment `extension=sqlite3` and set 
 * the path to the SQLite3 extensions directory:
 *      ```
 *      [sqlite3]
 *      ; Directory pointing to SQLite3 extensions
 *      ; https://php.net/sqlite3.extension-dir
 *      sqlite3.extension_dir = <PATH>
 *      ```
 */

$db = new SQLite3(':memory:');

// Name of the  compiled extension (.dylib for macOS/iOS, .so for linux/Android, .dll for Windows).
// The extension file must be located in the directory specified in the configure option `sqlite3.extension_dir`.
// https://www.php.net/manual/en/sqlite3.loadextension.php
$loaded = $db->loadExtension('js.so');
print_r("Extension loaded: " . ($loaded ? "true" : "false") . "\n");

$result = $db->query("SELECT js_version();");
while ($row = $result->fetchArray()) {
    print_r($row);
}
